src/dos/cdrom_image.cpp

```diff
@@ -959,12 +1095,14 @@ bool CDROM_Interface_Image::AddTrack(Track &curr, int &shift, int prestart, int
 bool CDROM_Interface_Image::HasDataTrack(void)
 {
 	//Data track has attribute 0x40
-	for(track_it it = tracks.begin(); it != tracks.end(); it++) {
-		if ((*it).attr == 0x40) {
-			return true;
+	bool result = false;
+	for(track_iter track = tracks.begin(); track != tracks.end(); track++) {
+		if (track->attr == 0x40) {
+			result = true;
+			break;
 		}
 	}
-	return false;
+	return result;
```

Comment:
for (const auto &track : tracks) {
        if (track.attr = 0x40)
            return true;
    }
    return false;

?

(this code asks for functional programming one-liner, alas - this is C++)

Reply:
Yes, that's more obvious - fixed in 91930311